stdlib/public/RemoteInspection/TypeRefBuilder.cpp

```diff
@@ -75,9 +80,15 @@ TypeRefBuilder::ReflectionTypeDescriptorFinder::
       ReflectionInfoIndexesSortedByTypeReferenceRange.begin(),
       ReflectionInfoIndexesSortedByTypeReferenceRange.end(), remoteAddr,
       [&](uint32_t ReflectionInfoIndex, remote::RemoteAddress remoteAddr) {
-        return ReflectionInfos[ReflectionInfoIndex]
-                   .TypeReference.endAddress()
-                   .getRemoteAddress() <= remoteAddr;
+        auto reflectionInfoAddress = ReflectionInfos[ReflectionInfoIndex]
+                                         .TypeReference.endAddress()
+                                         .getRemoteAddress();
+
+        if (reflectionInfoAddress.getAddressSpace() !=
+            remoteAddr.getAddressSpace())
+          return reflectionInfoAddress.getAddressSpace() <=
+                 remoteAddr.getAddressSpace();
```

Comment:
this seems unusual, I wouldn't think of address spaces as having an ordering. I can guess why, but maybe a comment.